nixos/modules/virtualisation/amazon-image.nix

```diff
@@ -33,23 +33,28 @@ in
 
     boot.growPartition = cfg.hvm;
 
-    fileSystems."/" = {
+    fileSystems."/" = mkIf (!cfg.zfs.enable) {
       device = "/dev/disk/by-label/nixos";
       fsType = "ext4";
       autoResize = true;
     };
 
-    fileSystems."/boot" = mkIf cfg.efi {
+    fileSystems."/boot" = if cfg.efi then {
       device = "/dev/disk/by-label/ESP";
       fsType = "vfat";
-    };
+    } else if cfg.zfs.enable then {
+      device = "boot/boot";
+      fsType = "zfs";
+    } else {};
+
+    boot.zfs.devNodes = mkIf cfg.zfs.enable "/dev/";
 
     boot.extraModulePackages = [
       config.boot.kernelPackages.ena
     ];
     boot.initrd.kernelModules = [ "xen-blkfront" "xen-netfront" ];
     boot.initrd.availableKernelModules = [ "ixgbevf" "ena" "nvme" ];
-    boot.kernelParams = mkIf cfg.hvm [ "console=ttyS0" "random.trust_cpu=on" ];
+    boot.kernelParams = mkIf cfg.hvm [ "console=ttyS0,115200n8" "random.trust_cpu=on" ];
```

Comment:
(Why) is this necessary?

Reply:
Ah, this is actually fixing a long-standing issue with our AWS AMIs where the console doesn't work.

Reply:
IMHO, it should be a separate commit with a message to that effect, then.

Reply:
Or, even better, a separate PR that will be relatively uncontroversial and likely merge sooner than this PR ;)

Reply:
Resolved in #134981.